### PR TITLE
feat: Cart-Management

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features added
- add to cart 
- remove from cart
- after clicking on `add to cart` the button changes to `go to cart`, and the redirects to cart
- increase decrease quantity from cart page
- if quantity is less than 1, it removes the product from cart
- shows the total price, discount and net total price
- number of items in the cart gets updated in the Navbar

# demo
https://user-images.githubusercontent.com/54719132/160653087-85cb1e8c-2ff0-466b-b197-d80f775eb51e.mp4

### CODE REVIEW

File: gozo-stores/src/components/Card.jsx [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
         <div className="card-component">
 
             <div class="product-card bg-white">
                 <div class="product-image-container">
                     <img class="product-image" src={props.src} alt={props.name} />
-                    <span class="like-product bg-gray"><img src="../../assets/icons/heart-outline.png" alt="like" /></span>
+                    <span class="like-product bg-gray"><img src="../../assets/icons/heart-outline.png" alt="like" /></span> 
                 </div>
                 <div class="product-card-body">
                     <div class="product-name"> {props.name} <span>({props.rating})</span></div>
```
Comment (PrashanthreddyAinala) [inline gozo-stores/src/components/Card.jsx:24]: what is the use of Parenthesis in this line I think it is not required.

File: gozo-stores/src/components/HorizontalCard.jsx [hunk 2]
```diff
@@ -9,14 +15,33 @@ export function HorizontalCard(props) {
...
+                            deleteFromCart(token, props.productId)
+                        }
+                        else {
+                            updateCartQuantity(token, props.productId, "decrement")}
+                        }
+                    }>-</button> 
+
+                    <span>{props.quantity}</span> 
+
+                    <button onClick={() => updateCartQuantity(token, props.productId, "increment")}>+</button>
```
Comment (PrashanthreddyAinala) [inline gozo-stores/src/components/HorizontalCard.jsx:40]: For good practice you can pass the payload with uppercase letter it's just a suggestion not mandatory.

**Review by PrashanthreddyAinala — COMMENTED** [review]
Rest all code looks fine for me and great UI.

File: gozo-stores/src/context/cart-context.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+const CartContext = createContext();
+
+function CartProvider({children}) {
+
+    const [myCart, setMyCart] = useState([])
+
+    const addToCart = async (token, product) => {
+      try {
+        const res = await axios.post('/api/user/cart', {product}, {
```
Comment (anupkgurung) [inline gozo-stores/src/context/cart-context.js:12]: =>try to use proper names res i assume is for response so be it response.
=>you can destructure the response here directly const {data : { cart}}

**Review by anupkgurung — APPROVED** [review]
Nice use of comments to describe it's use.
clean and simple code

File: gozo-stores/src/components/Card.jsx (hunk 1 above)
Comment (maitrakhatri) [reply gozo-stores/src/components/Card.jsx:24]: it's not required but its a habit to use () after return to make sure I'm returning only one element

File: gozo-stores/src/components/HorizontalCard.jsx (hunk 2 above)
Comment (maitrakhatri) [reply gozo-stores/src/components/HorizontalCard.jsx:40]: I'll keep it in mind

**Review by PrashanthreddyAinala — APPROVED** [review]

File: gozo-stores/src/context/cart-context.js (hunk 3 above)
Comment (maitrakhatri) [reply gozo-stores/src/context/cart-context.js:12]: I'll take care of both things

**Pull request merged**